### PR TITLE
chore(Makefile): add test-race target

### PR DESCRIPTION
**Pull request by johnstcn**
Adds a Makefile target to run tests with the race detector enabled.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -668,6 +668,11 @@ test-postgres-docker:
 	done
 .PHONY: test-postgres-docker
 
+# Make sure to keep this in sync with test-go-race from .github/workflows/ci.yaml.
+test-race: test-clean
```
Comment (mafredri) [inline Makefile:672]: Why do we wipe the test cache when we run via Makefile? Adding `-count=1` should work the same and wouldn't trash the cache (e.g. for non-race tests).

**Review by mafredri — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (johnstcn) [reply Makefile:672]: Should we update existing invocations to use `-count=1` as well?

Comment (johnstcn) [reply Makefile:672]: Actually, I'll make a separate PR for that and we can debate in there.

Comment (mafredri) [reply Makefile:672]: Good suggestion, I think that would be preferable 👍🏻

**Review by mtojek — APPROVED** [review]
👍

**Pull request merged**